js/packages/core/src/Reconciler.res

```diff
@@ -87,14 +87,13 @@ let rec visit = (
 }
 
 @genType
-let renderWithDelegate = (delegate, graphs) => {
+let renderWithDelegate = (delegate, graphs, rootFadeInMs, rootFadeOutMs) => {
   let visitSet = Set.make()
   let roots = Belt.List.mapWithIndex(Belt.List.fromArray(graphs), (i, g) => {
     NodeRepr.create("root", {"channel": i}, [g])
   })
 
   visit(delegate, visitSet, roots)
-
-  RenderDelegate.activateRoots(delegate, Belt.List.toArray(Belt.List.map(roots, r => r.hash)))
+  RenderDelegate.activateRoots(delegate, Belt.List.toArray(Belt.List.map(roots, r => r.hash)), rootFadeInMs, rootFadeOutMs)
```

Comment:
I'd prefer to keep the custom fade times out of the activateRoots chain, just because we already have a suitable pattern for propagating properties like this to the nodes that need them 3 lines up–

```
NodeRepr.create("root", {"channel": i, "fadeInMs": rootFadeInMs, "fadeOutMs": rootFadeOutMs}, [g])
```

With just that change we can keep `activateRoots` totally ignorant to the business of root node fades, which I think is valuable (any part of the code that doesn't *need* to know something, shouldn't know it). Plus this way we don't change anything about the instruction format

Reply:
oh nice! didn't realize you could pass properties into `create`. I like this much better 👍